app/renderer/components/Session/CapabilityControl.js

```diff
@@ -7,24 +7,24 @@ import { INPUT } from '../AntdTypes';
 
 const {dialog} = remote;
 
-
 export default class NewSessionForm extends Component {
 
-  getLocalFilePath (success) {
-    dialog.showOpenDialog((filepath) => {
-      if (filepath) {
-        success(filepath);
+  async getLocalFilePath (success) {
+    await dialog.showOpenDialog({
+      properties: ['openFile'],
+    }).then((files) => {
+      if (files.filePaths) {
+        success(files.filePaths);
       }
-    });
-    this.handleSetType = this.handleSetType.bind(this);
+    }).catch(/* ignore */);
   }
 
   render () {
     const {cap, onSetCapabilityParam, isEditingDesiredCaps, id, t} = this.props;
 
     const buttonAfter = <FileOutlined
       className={SessionStyles['filepath-button']}
-      onClick={() => this.getLocalFilePath((filepath) => onSetCapabilityParam(filepath[0]))} />;
+      onClick={() => this.getLocalFilePath((filePaths) => onSetCapabilityParam(filePaths[0]))} />;
```

Comment:
nit: I personally find it hard to read when functions like these are being passed as arguments. It would be easier to parse this piece of the code if we had something like

```
onClick={async () => {
  const filePaths = await this.getLocalFilePath();
  if (filePaths) {
    onSetCapabilityParam(filePaths[0]);
  }
} 
```